comm.go

```diff
@@ -121,6 +122,16 @@ func (p *PubSub) handleNewPeer(ctx context.Context, pid peer.ID, outgoing <-chan
 	}
 }
 
+func (p *PubSub) handleNewPeerWithBackoff(ctx context.Context, pid peer.ID, outgoing <-chan *RPC) {
+	delay := p.deadPeerBackoff.updateAndGet(pid)
```

Comment:
I think we need to add a failure more if we have backed off too much and simply give up; say we try up to 10 times and then `updateAndGet` returns an error and we close the channel and forget the peer.

How does that sound?

Reply:
maybe 10 is even too much, 3-4 attempts should be enough.

Reply:
Done